packages/twenty-server/src/modules/workflow/common/workspace-services/workflow-common.workspace-service.ts

```diff
@@ -55,4 +57,35 @@ export class WorkflowCommonWorkspaceService {
 
     return { ...workflowVersion, trigger: workflowVersion.trigger };
   }
+
+  async cleanWorkflowsSubEntities(workflowIds: string[]): Promise<void> {
+    const workflowVersionRepository =
+      await this.twentyORMManager.getRepository<WorkflowVersionWorkspaceEntity>(
+        'workflowVersion',
+      );
+
+    const workflowRunRepository =
+      await this.twentyORMManager.getRepository<WorkflowRunWorkspaceEntity>(
+        'workflowRun',
+      );
+
+    const workflowEventListenerRepository =
+      await this.twentyORMManager.getRepository<WorkflowEventListenerWorkspaceEntity>(
+        'workflowEventListener',
+      );
+
+    workflowIds.map(async (workflowId) => {
```

Comment:
I guess if you really want to fire-and-forget, you don't need map here but foreach instead. This won't guarantee the execution of your promises though I'd suggest using Promise.all()